### PR TITLE
update export, rsync cmds

### PR DESCRIPTION
**Pull request by juanvallejo**
Updates export and rsync commands to follow updated upstream patterns

cc @deads2k

### CODE REVIEW

File: pkg/oc/cli/cmd/export.go [hunk 1]
```diff
@@ -52,9 +50,41 @@ var (
 	  %[1]s export service -o json`)
 )
 
+type ExportOptions struct {
+	genericclioptions.IOStreams
```
Comment (deads2k) [inline pkg/oc/cli/cmd/export.go]: nit: bottom of struct

File: pkg/oc/cli/cmd/export.go [hunk 2]
```diff
@@ -63,64 +93,76 @@ func NewCmdExport(fullName string, f kcmdutil.Factory, streams genericclioptions
...
+
+	cmd.Flags().StringVar(&o.AsTemplateName, "as-template", o.AsTemplateName, "Output a Template object with specified name instead of a List or single object.")
+	cmd.Flags().BoolVar(&o.Exact, "exact", o.Exact, "If true, preserve fields that may be cluster specific, such as service clusterIPs or generated names")
+	cmd.Flags().BoolVar(&o.Raw, "raw", o.Raw, "If true, do not alter the resources in any way after they are loaded.")
+	cmd.Flags().StringVarP(&o.Selector, "selector", "l", o.Selector, "Selector (label query) to filter on")
+	cmd.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", o.AllNamespaces, "If true, list the requested object(s) across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
 	cmd.Flags().Bool("all", true, "DEPRECATED: all is ignored, specifying a resource without a name selects all the instances of that resource")
 	cmd.Flags().MarkDeprecated("all", "all is ignored because specifying a resource without a name selects all the instances of that resource")
-	kcmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().StringVar(&o.OutputVersion, "output-version", o.OutputVersion, "The preferred API versions of the output objects")
```
Comment (deads2k) [inline pkg/oc/cli/cmd/export.go]: This leaps out and stabs you.  Issue to remove it.  It isn't consistent.

Comment (deads2k) [reply pkg/oc/cli/cmd/export.go]: Actually, mark deprecated, indicate it's ignored.  You don't even use it anymore.

File: pkg/oc/cli/cmd/rsync/rsync.go [hunk 3]
```diff
@@ -76,6 +76,8 @@ type podChecker interface {
 
 // RsyncOptions holds the options to execute the sync command
 type RsyncOptions struct {
+	genericclioptions.IOStreams
```
Comment (deads2k) [inline pkg/oc/cli/cmd/rsync/rsync.go]: bottom

**Comment by deads2k** [comment]
nits.  lgtm otherwise.  self tag when nits are fixed

/approve

**Comment by juanvallejo** [comment]
nits fixed. adding tag

**Comment by stevekuznetsov** [comment]
@deads2k @juanvallejo we may remove the ability for everyone to add labels in the future -- so this https://github.com/openshift/origin/pull/20271#issuecomment-403918516 won't work. In upstream we commonly use `/hold` (by David in this example) to allow others (Juan) to `/hold cancel` later without privileges to the repo

**Pull request merged**